search-commons/src/main/java/no/unit/nva/search2/common/QueryTools.java

```diff
@@ -0,0 +1,235 @@
+import static nva.commons.core.paths.UriWrapper.fromUri;
+import java.net.URI;
+import java.net.URLDecoder;
+import java.nio.charset.StandardCharsets;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+import no.unit.nva.search2.constant.Defaults;
+import no.unit.nva.search2.constant.Words;
+import no.unit.nva.search2.enums.ParameterKey;
+import no.unit.nva.search2.enums.ParameterKey.ParamKind;
+import nva.commons.core.JacocoGenerated;
+import org.apache.lucene.search.join.ScoreMode;
+import org.opensearch.index.query.MatchQueryBuilder;
+import org.opensearch.index.query.MultiMatchQueryBuilder;
+import org.opensearch.index.query.MultiMatchQueryBuilder.Type;
+import org.opensearch.index.query.Operator;
+import org.opensearch.index.query.QueryBuilder;
+import org.opensearch.index.query.QueryBuilders;
+import org.opensearch.search.sort.SortOrder;
+
+public final class QueryTools<K extends Enum<K> & ParameterKey> {
+
+    public static Boolean valueToBoolean(String value) {
+        return ONE.equals(value) ? Boolean.TRUE : Boolean.valueOf(value);
+    }
+
+    public static String decodeUTF(String encoded) {
+        return URLDecoder.decode(encoded, StandardCharsets.UTF_8);
+    }
+
+    public static Collection<Entry<String, String>> queryToMapEntries(URI uri) {
+        return queryToMapEntries(uri.getQuery());
+    }
+
+    public static Collection<Entry<String, String>> queryToMapEntries(String query) {
+        return
+            nonNull(query)
+                ? Arrays.stream(query.split(AMPERSAND))
+                .map(s -> s.split(EQUAL))
+                .map(QueryTools::stringsToEntry)
+                .toList()
+                : Collections.emptyList();
+    }
+
+    public static Entry<String, String> stringsToEntry(String... strings) {
+        return new Entry<>() {
+            @Override
+            public String getKey() {
+                return strings[0];
+            }
+
+            @Override
+            public String getValue() {
+                return attempt(() -> strings[1]).orElse((f) -> EMPTY_STRING);
+            }
+
+            @Override
+            @JacocoGenerated
+            public String setValue(String value) {
+                return null;
+            }
+        };
+    }
+
+    public static Entry<String, SortOrder> entryToSortEntry(Entry<String, String> entry) {
+        return new Entry<>() {
+            @Override
+            public String getKey() {
+                return entry.getKey();
+            }
+
+            @Override
+            public SortOrder getValue() {
+                var sortOrder = nonNull(entry.getValue()) && !entry.getValue().isEmpty()
+                    ? entry.getValue()
+                    : Defaults.DEFAULT_SORT_ORDER;
+                return SortOrder.fromString(sortOrder);
```

Comment:
Maybe something like this?
`return sortOrderIsPresent(entry) ? SortOrder.fromString(entry.getValue()) : defaultSortOrder()`